infra/src/main/kotlin/fi/fta/geoviite/infra/geometry/VerticalGeometryListing.kt

```diff
@@ -177,6 +179,96 @@ fun toVerticalGeometryListing(
+        formatTo3Decimals(it.start.height)
+    },
+    CsvEntry(translateVerticalGeometryListingHeader(VerticalGeometryListingHeader.ANGLE_START)) {
+        it.start.angle?.let(::formatTo6Decimals)
+    },
+    CsvEntry(translateVerticalGeometryListingHeader(VerticalGeometryListingHeader.TRACK_ADDRESS_POINT)) {
+        it.point.address?.let { address ->
+            formatTrackMeter(
+                address.kmNumber,
+                address.meters
+            )
+        }
+    },
+    CsvEntry(translateVerticalGeometryListingHeader(VerticalGeometryListingHeader.HEIGHT_POINT)) {
+        formatTo3Decimals(it.point.height)
+    },
+    CsvEntry(translateVerticalGeometryListingHeader(VerticalGeometryListingHeader.TRACK_ADDRESS_END)) {
+        it.end.address?.let { address ->
+            formatTrackMeter(
+                address.kmNumber,
+                address.meters
+            )
+        }
+    },
+    CsvEntry(translateVerticalGeometryListingHeader(VerticalGeometryListingHeader.HEIGHT_END)) {
+        formatTo3Decimals(it.end.height)
+    },
+    CsvEntry(translateVerticalGeometryListingHeader(VerticalGeometryListingHeader.ANGLE_END)) {
+        it.end.angle?.let(::formatTo6Decimals)
+    },
+    CsvEntry(translateVerticalGeometryListingHeader(VerticalGeometryListingHeader.RADIUS)) { it.radius },
+    CsvEntry(translateVerticalGeometryListingHeader(VerticalGeometryListingHeader.TANGENT)) {
+        it.tangent?.let(
+            ::formatTo3Decimals
+        )
+    },
+    CsvEntry(translateVerticalGeometryListingHeader(VerticalGeometryListingHeader.LINEAR_SECTION_BACKWARD_LENGTH)) {
+        it.linearSectionBackward.stationValueDistance?.let(
+            ::formatTo3Decimals
+        )
+    },
+    CsvEntry(translateVerticalGeometryListingHeader(VerticalGeometryListingHeader.LINEAR_SECTION_BACKWARD_LINEAR_SECTION)) {
+        it.linearSectionBackward.linearSegmentLength?.let(
+            ::formatTo3Decimals
+        )
+    },
+    CsvEntry(translateVerticalGeometryListingHeader(VerticalGeometryListingHeader.LINEAR_SECTION_FORWARD_LENGTH)) {
+        it.linearSectionForward.stationValueDistance?.let(
+            ::formatTo3Decimals
+        )
+    },
+    CsvEntry(translateVerticalGeometryListingHeader(VerticalGeometryListingHeader.LINEAR_SECTION_FORWARD_LINEAR_SECTION)) {
+        it.linearSectionForward.linearSegmentLength?.let(
+            ::formatTo3Decimals
+        )
+    },
+    CsvEntry(translateVerticalGeometryListingHeader(VerticalGeometryListingHeader.STATION_START)) { formatTo3Decimals(it.start.station) },
+    CsvEntry(translateVerticalGeometryListingHeader(VerticalGeometryListingHeader.STATION_POINT)) { formatTo3Decimals(it.point.station) },
+    CsvEntry(translateVerticalGeometryListingHeader(VerticalGeometryListingHeader.STATION_END)) { formatTo3Decimals(it.end.station) },
+)
+
+private fun formatTo3Decimals(number: Double) =
```

Comment:
Meillä on jo geneeriset pyöristysfunkkarit joten niitä kannattaa ehkä käyttää. En ole varma miten päin format muutenkaan pyöristää. Nuo valmiit toki palauttaa BigDecimal-tyypin mutta sille voi vain sanoa toString

Reply:
Itse asiassa, nuo tyypit voisi olla tuolla listing-oliossa jo bigdecimal, jolloin ne sisältää tiedon tarkkuudesta ja täällä ei tarvitse tehdä pyöristyksiä ollenkaan (oliota luodessa pistetään haluttu bigdecimal tarkkuus käyttämällä round-funkkaria). Niin se taitaa olla elementtipuolella.

Reply:
Elementtilistaus käyttää `.toPlainString()`:iä. Pystygeometrialistaukseen haluttiin nämä eksplisiittiset tarkkuudet, joten siksi nämä format-funkkarit täällä. Mutta hyvä pointti tuosta `BigDecimal`ista, voisin muutta nämä toimimaan sillä